research/2019-12-16_searching_with_the_calm_api.md

```diff
@@ -0,0 +1,241 @@
+# Searching with the CALM API
+
+### 16 December 2019, AWLC
+
+This has some notes on using the CALM API, and an example script for searching all the records in the CALM catalogue.
+
+
+
+## Motivation
+
+As part of the born-digital content migration, I had a collection of directories with accession-like numbers, and I wanted to get a list of all the accession numbers in CALM, so I could match the directories to accession numbers:
+
+    AAUICW_2185 ~> AAUICW/2185 ?
+    AAUNAM_2213 ~> AAUNAM/2213 ?
+    AAUNAM_2254 ~> AAUNAM/2254 ?
+
+Exporting a list of accession numbers from the CALM GUI would have been tricky (I don't have access to CALM on my Mac), so I tried using the CALM API to export the data.
+
+<details>
+  <summary>What is an accession/accession number?</summary>
+
+  > In library science, an *accession* is all the material that gets collected at one time from a single source.
+  > The *accession identifier* is the unique identifier given to that batch of material.
+  >
+  > At Wellcome, the accession identifier is made of two parts: the *accession number* (sequential, currently up to 2551) and a *collection reference* (such as WT, PPMTW, SANCT).
+  > The two are written together, separated by a slash.
+  > Examples:
+  >
+  > The accession number is globally unique: if there's an accession `WT/123`, you won't see `123` with any other collection reference.
+</details>
+
+
+
+## Resources
+
+*   The docs for Wellcome's CALM API installation are available at <http://archives.wellcomelibrary.org/CalmAPI/help/index.htm> (externally visible) or <http://wt-calm/CalmAPI/help/index.htm> (internal only).
+
+*   You can check the status of the API at <http://wt-calm/CalmAPI/>.
+
+*   Talk to Library Systems for credentials to access the API.
+
+
+
+## Searching the API
+
+Useful pieces for searching the API:
+
+*   **Make a Search request using the SOAP API.**
+
+    Example request/responses are here: <http://archives.wellcomelibrary.org/CalmAPI/help/example_search.htm>
+    This returns an XML response that tells you how many results there are.
+
+    I know you can search the "Catalog", "Accessn" and "Locations" databases; there might be more I'm unaware of.
```

Comment:
There are lots more but not all of them are populated / relevant. Of those which are "Persons" is the most significant - contains authority records for people and functions, most notably in relation to the Wellcome Trust corporate archive.

Reply:
Indeed, "Persons" does return a bunch of search results!